pkg/pillar/pubsub/legacy/legacy.go

```diff
@@ -10,51 +10,102 @@ var (
 	defaultPubsub = pubsub.New(&socketdriver.SocketDriver{})
 )
 
-// Publish create a `Publication` for the given agent name and topic type.
+// Publish create a `NewPublication` for the given agent name and topic type.
 func Publish(agentName string, topicType interface{}) (pubsub.Publication, error) {
 	log.Infof("legacy.Publish agentName(%s)", agentName)
-	return defaultPubsub.Publish(agentName, topicType)
+	return defaultPubsub.NewPublication(&pubsub.PublicationOptions{
+		AgentName: agentName,
+		TopicType: topicType,
+	})
 }
 
-// PublishPersistent create a `Publication` for the given agent name and topic
+// PublishPersistent create a `NewPublication` for the given agent name and topic
 // type, but with persistence of the messages across reboots.
 func PublishPersistent(agentName string, topicType interface{}) (pubsub.Publication, error) {
 	log.Infof("legacy.PublishPersistent agentName(%s)", agentName)
-	return defaultPubsub.PublishPersistent(agentName, topicType)
+	return defaultPubsub.NewPublication(&pubsub.PublicationOptions{
+		AgentName:  agentName,
+		TopicType:  topicType,
+		Persistent: true,
+	})
 }
 
-// PublishScope create a `Publication` for the given agent name and topic,
+// PublishScope create a `NewPublication` for the given agent name and topic,
```

Comment:
Same as before, revert the comment. It is creating a `Publication` (the thing it is returning), not a `NewPublication` (the func call)